org.jacoco.core/src/org/jacoco/core/internal/runtime/ClassInjectionRuntime.java

```diff
@@ -0,0 +1,297 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2018 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.runtime;
```

Comment:
Why is this implementation in a private package while the other runtime implementations are in `org.jacoco.core.runtime`? Would move this to the same package.

Reply:
@marchof to be honest I don't like that existing implementations of `IRuntime` are in public package - IMO they are implementation details of agent. But ok - can move it.
BTW `org.jacoco.agent.rt.internal.Offline` is in internal package 😉

Reply:
Another option would be to move all runtimes to internal packages of the agent. On the other hand the current public API allows to use runtimes outside the agent (like in the `CoreTutorial` example).

For now I would move it next to the other implementations and we can discuss this in a separate PR, ok?

Reply:
Done.